pipelinedp4j/tests/com/google/privacy/differentialprivacy/pipelinedp4j/spark/SparkEncodersTest.kt

```diff
@@ -89,12 +89,14 @@ class SparkEncodersTest {
         val inputEncoder =
             sparkEncoderFactory.tuple2sOf(sparkEncoderFactory.strings(), sparkEncoderFactory.ints()).encoder
 
-        val dataset = spark.createDataset(input, inputEncoder)
+        val dataset = sparkSession.spark.createDataset(input, inputEncoder)
         assertThat(dataset.collectAsList()).containsExactlyElementsIn(input)
     }
 
     companion object {
-        private val spark: SparkSession = createSparkSession()
+        @JvmField
+        @ClassRule
+        val sparkSession = SparkSessionRule()
```

Comment:
just to double check: do you know whether it is ok when SparkSession is shared across multiple tests?

Reply:
Then should we make it not a class rule but just rule? Then it will be called before and after each test.

Reply:
Since we are not modifying any spark property, it is fine to keep Class Rule since SparkSession creation is expensive operation.